invenio/b2share/modules/b2deposit/restful.py

```diff
@@ -205,9 +206,14 @@ def get(self, oauth, domain_name, **kwargs):
         # get domain id from domain name
         domain = Bib98x.query.filter_by(value=domain_name).first()
         if domain is None:
-            abort(404, status=404,
-                  message="Please try a valid domain name: " +
-                         ", ".join(metadata_classes().keys()))
+            if domain_name not in metadata_classes().keys():
```

Comment:
I think it’s ok. Previously, if user asked for an empty domain he was getting the message for entering wrong domain name, which is not the correct message. Instead I think he should get an empty list. Now if domain is “None”, it checks for two cases: if it is not in the domain list (metadata_classes().keys()) then he’ll get an error message, else he will get an empty list.